internal/controller/dscinitialization/servicemesh.go

```diff
@@ -0,0 +1,217 @@
+package dscinitialization
+
+import (
+	"context"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	k8serr "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+
+	dsciv1 "github.com/opendatahub-io/opendatahub-operator/v2/api/dscinitialization/v1"
+	serviceApi "github.com/opendatahub-io/opendatahub-operator/v2/api/services/v1alpha1"
+	"github.com/opendatahub-io/opendatahub-operator/v2/internal/controller/status"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/conditions"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/resources"
+)
+
+func (r *DSCInitializationReconciler) handleServiceMesh(ctx context.Context, dscInit *dsciv1.DSCInitialization) error {
+	log := logf.FromContext(ctx)
+
+	if dscInit.Spec.ServiceMesh == nil {
+		log.Info("ServiceMesh not defined in DSCI, deleting ServiceMesh CR if present")
+		return r.deleteServiceMesh(ctx)
+	}
+
+	conditions := dscInit.Status.Conditions
+	switch dscInit.Spec.ServiceMesh.ManagementState {
+	case operatorv1.Managed:
+		log.Info("ServiceMesh set to Managed in DSCI, attempting to update ServiceMesh CR")
+		desiredServiceMesh := &serviceApi.ServiceMesh{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: serviceApi.ServiceMeshInstanceName,
+			},
+			Spec: serviceApi.ServiceMeshSpec{
+				ControlPlane: serviceApi.ServiceMeshControlPlaneSpec{
+					Name:              dscInit.Spec.ServiceMesh.ControlPlane.Name,
+					Namespace:         dscInit.Spec.ServiceMesh.ControlPlane.Namespace,
+					MetricsCollection: dscInit.Spec.ServiceMesh.ControlPlane.MetricsCollection,
+				},
+				Auth: serviceApi.ServiceMeshAuthSpec{
+					Namespace: dscInit.Spec.ServiceMesh.Auth.Namespace,
+					Audiences: dscInit.Spec.ServiceMesh.Auth.Audiences,
+				},
+			},
+		}
+
+		if err := controllerutil.SetControllerReference(dscInit, desiredServiceMesh, r.Client.Scheme()); err != nil {
+			return err
+		}
+
+		err := resources.Apply(
+			ctx,
+			r.Client,
+			desiredServiceMesh,
+			client.FieldOwner(fieldManager),
+			client.ForceOwnership,
+		)
+		if err != nil && !k8serr.IsAlreadyExists(err) {
```

Comment:
`IsAlreadyExists` is probably not required as SSA would create or update the object